src/test/java/org/mockito/internal/configuration/plugins/PluginLoaderTest.java

```diff
@@ -0,0 +1,88 @@
+package org.mockito.internal.configuration.plugins;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.internal.util.io.IOUtil;
+import org.mockito.plugins.PluginSwitch;
+import org.mockitoutil.ClassLoaders;
+import org.mockitoutil.TestBase;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.anyString;
+
+/**
+ *  Test PluginLoader for case when Plugin class is loaded via class loader which is different from current context
+ *  class loader.
+ */
+public class PluginLoaderTest extends TestBase {
+
+    @InjectMocks
+    private PluginLoader pluginLoader;
+
+    @Mock
+    private PluginSwitch pluginSwitch;
+
+    @Before
+    public void setUp() {
+        given(pluginSwitch.isEnabled(anyString())).willReturn(true);
+    }
+
+    @Test
+    public void without_custom_context_class_loader_plugin_should_be_load() {
+
+
+        Object pluginInterface = pluginLoader.loadPlugin(PluginInterface.class, "org.mockito.internal" +
+                                                                                                  ".configuration.plugins.DefaultPlugin");
+
+        assertEquals(PluginImpl.class, pluginInterface.getClass());
+    }
+
+    @Test
+    public void if_custom_context_class_loader_is_used_then_plugin_should_be_load() throws Exception {
+
+        ClassLoader classLoader = ClassLoaders.inMemoryClassLoader().withParent(getParentClassLoader()).withClassDefinition("", readResource()).build();
```

Comment:
If you look at source code of  `InMemoryClassLoader`

 ```
@Override
        public Enumeration<URL> getResources(String ignored) throws IOException {
            return inMemoryOnly();
        }

        private Enumeration<URL> inMemoryOnly() {
            final Set<String> names = inMemoryClassObjects.keySet();
            return new Enumeration<URL>() {
                private final MemHandler memHandler = new MemHandler(InMemoryClassLoader.this);
                private final Iterator<String> it = names.iterator();

                public boolean hasMoreElements() {
                    return it.hasNext();
                }

                public URL nextElement() {
                    try {
                        return new URL(null, SCHEME + ":" + it.next(), memHandler);
                    } catch (MalformedURLException rethrown) {
                        throw new IllegalStateException(rethrown);
                    }
                }
            };
        }
```

The `InMemoryClassLoader` returns all class as resources and `PluginLoader` reads these resources. So if I remove `readResource` then `PluginLoader` will not find any plugin and then create default instance. 

I can replace `readResource` via `"org.mockito.internal.configuration.plugins.PluginImpl".getBytes()`, but I prefer use same arrangement as for another test.